src/main/java/de/rabea/controller/AssetController.java

```diff
@@ -1,34 +1,41 @@
 package de.rabea.controller;
 
-import de.rabea.ContentStorage;
 import de.rabea.Controller;
 import de.rabea.EntityTagChecker;
 import de.rabea.request.Directory;
 import de.rabea.request.HttpRequest;
 import de.rabea.response.HttpResponse;
 
-import static de.rabea.response.head.StatusLine.NO_CONTENT;
-import static de.rabea.response.head.StatusLine.OK;
+import static de.rabea.response.head.StatusLine.*;
 
 public class AssetController extends Controller {
-
     private final Directory directory;
-    private final ContentStorage contentStorage;
 
-    public AssetController(Directory directory, ContentStorage contentStorage) {
+    public AssetController(Directory directory) {
         this.directory = directory;
-        this.contentStorage = contentStorage;
     }
 
     @Override
     public HttpResponse doGet(HttpRequest request) {
-        contentStorage.storeFileContent(directory.contentOfFile(request.requestLine().uri()));
-        return new HttpResponse(OK, contentStorage.content());
+        String file = request.requestLine().uri();
+        if (containsRange(request)) {
+           return new HttpResponse(PARTIAL_CONTENT, directory.partialFileContent(file, getRange(request)));
+        }
+        return new HttpResponse(OK, directory.fileContent(file));
+    }
+
+    private String getRange(HttpRequest request) {
+        return request.requestHeaders().get("Range");
+    }
+
+    private boolean containsRange(HttpRequest request) {
+        return request.requestHeaders().containsKey("Range");
     }
 
     @Override
     public HttpResponse doPatch(HttpRequest request) {
-        if (EntityTagChecker.isCorrectTag(directory.contentOfFile(request.requestLine().uri()), request.requestHeaders().get("If-Match"))) {
+        if (EntityTagChecker.isCorrectTag(directory.fileContent(request.requestLine().uri()),
+                request.requestHeaders().get("If-Match"))) {
```

Comment:
I do think breaking up that line into two is a good idea, but can you think of another way to give this condition more expressiveness?